atomicfu/src/jsAndWasmSharedMain/kotlin/kotlinx/atomicfu/parking/KThread.kt

```diff
@@ -0,0 +1,17 @@
+package kotlinx.atomicfu.parking
+
+actual class KThread internal actual constructor() {
+    actual companion object {
+        actual fun currentThread() = thisKthread
+    }
+}
+
+actual class Parker actual private constructor() {
+    actual companion object {
+        actual fun park() {}
+        actual fun parkNanos(nanos: Long) {}
+        actual fun unpark(kThread: KThread) {}
+    }
+}
+private val thisKthread = KThread()
+actual fun currentThreadId(): Long = 1
```

Comment:
If we do decide to avoid spurious wake-ups, this implementation is incorrect. But in any case, I think it's also useless to provide it: I'm not sure what anyone using thread synchronization on single-threaded runtimes even expects. As part of the mutex implementation, this particular implementation makes sense, but not for other primitives.

Proposal: create a source set combining JVM and Native (in `kotlinx.coroutines`, we call it `concurrent`), and instead of providing parking in common code, only provide it in `concurrent`. Also, with this source set, it's easier to share tests between `jvmTest` and `nativeTest`: for most of them, it seems possible to put them in `concurrentTest`.

Reply:
Good suggestion! Added the sourceSet.
About the tests, spawning threads on Jvm and Native have different API's correct? Or would you suggest creating an expect/actual constructinon for thread manipulation?

Reply:
We do `expect`/`actual` in our test source sets all the time, and it works without issues: https://github.com/Kotlin/kotlinx.coroutines/blob/master/kotlinx-coroutines-core/concurrent/test/ConcurrentTestUtilities.common.kt.

Reply:
Done